### PR TITLE
fix: apply component spec to trvy deployment

### LINKED ISSUE #436: The pull secret is not applied to trivy
I tried to deploy harborcluster using the cr:
https://github.com/goharbor/harbor-operator/blob/develop/manifests/samples/full_stack_fs.yaml

I set the imagePullSecret:
```
  imagePullSecrets:
      - name: registry-secret
```

But seems the setting is not applied to the pod of trivy:
```
root@42278b9601baa53d7db387727f0d9199 [ ~ ]# kubectl get po harbor-harbor-harbor-portal-58d96d6dd-gbnz7 -n harbor -o yaml|grep -i pullsecret
        f:imagePullSecrets:
  imagePullSecrets:
root@42278b9601baa53d7db387727f0d9199 [ ~ ]# kubectl get po harbor-harbor-harbor-trivy-67b8b5849b-tzr85 -n harbor -o yaml|grep -i pullsecret
root@42278b9601baa53d7db387727f0d9199 [ ~ ]#
```

### PR DESCRIPTION
**Pull request by heww**
Closes #436

Signed-off-by: He Weiwei <hweiwei@vmware.com>

### CODE REVIEW

File: manifests/samples/full_stack_fs.yaml [hunk 1]
```diff
@@ -73,6 +73,8 @@ spec:
     enabled: true
   portal: {}
   registry: {}
+  trivy:
```
Comment (steven-zou) [inline manifests/samples/full_stack_fs.yaml:76]: @heww 

why not 

```
trivy: {}
```

empty storage is required?

**Review by steven-zou — COMMENTED** [review]
a comment

File: manifests/samples/full_stack_fs.yaml (hunk 1 above)
Comment (heww) [reply manifests/samples/full_stack_fs.yaml:76]: `storage` is required, not optional

Comment (bitsf) [inline manifests/samples/full_stack_fs.yaml:76]: don't add trivy in this file, it should in full_stack_other_fs.yml

**Review by bitsf — CHANGES_REQUESTED** [review]

File: manifests/samples/full_stack_fs.yaml (hunk 1 above)
Comment (heww) [reply manifests/samples/full_stack_fs.yaml:76]: Why can't we enable trivy in this manifest?

**Review by bitsf — APPROVED** [review]

**Review by steven-zou — APPROVED** [review]
lgtm

**Pull request merged**